llvm/lib/Transforms/IPO/SampleProfile.cpp

```diff
@@ -1391,10 +1391,12 @@ SampleProfileLoader::shouldInlineCandidate(InlineCandidate &Candidate) {
       return InlineCost::getAlways("preinliner");
   }
 
-  // For old FDO inliner, we inline the call site as long as cost is not
-  // "Never". The cost-benefit check is done earlier.
+  // For old FDO inliner, we inline the call site if it is below hot threshold,
+  // even if the function is hot based on sample profile data. This is to
+  // prevent huge functions from being inlined.
   if (!CallsitePrioritizedInline) {
-    return InlineCost::get(Cost.getCost(), INT_MAX);
+    return InlineCost::get(Cost.getCost(),
+                           Params.HotCallSiteThreshold.value_or(INT_MAX));
```

Comment:
Should we use `SampleThreshold` here? and then simply make it unconditional, i.e. merge with line 1404.

Reply:
@huangjd any reason not to merge this with line 1403 in the latest version?

Reply:
It requires CallsitePrioritizedInline to set  SampleThreshold to SampleHotCallSiteThreshold, otherwise it is using SampleColdCallSiteThreshold. However according to flag description CallsitePrioritizedInline is only used by CSSPGO but not for context-free profiles, so given a context free FDO profile, we don't want to use SampleColdCallSiteThreshold if the callsite is hot